sources/packages/api/src/esdc-integration/e-cert-full-time-integration/e-cert-full-time-response.service.ts

```diff
@@ -0,0 +1,154 @@
+import { Injectable, LoggerService } from "@nestjs/common";
+import { InjectLogger } from "../../common";
+import {
+  DisbursementScheduleErrorsService,
+  DisbursementScheduleService,
+  ConfigService,
+} from "../../services";
+import { ProcessSftpResponseResult } from "./models/e-cert-full-time-integration.model";
+import { ECertFullTimeIntegrationService } from "./e-cert-full-time-integration.service";
+import { ECertResponseRecord } from "./e-cert-files/e-cert-response-record";
+import { getUTCNow } from "../../utilities";
+import { ESDCIntegrationConfig } from "../../types";
+
+@Injectable()
+export class ECertFullTimeResponseService {
+  private readonly esdcConfig: ESDCIntegrationConfig;
+
+  constructor(
+    config: ConfigService,
+    private readonly disbursementScheduleService: DisbursementScheduleService,
+    private readonly eCertFullTimeService: ECertFullTimeIntegrationService,
+    private readonly disbursementScheduleErrorsService: DisbursementScheduleErrorsService,
+  ) {
+    this.esdcConfig = config.getConfig().ESDCIntegration;
+  }
+
+  /**
+   * Download all files from E-Cert Response folder on SFTP and process them all.
+   * @returns Summary with what was processed and the list of all errors, if any.
+   */
+  async processResponses(): Promise<ProcessSftpResponseResult[]> {
+    const filePaths = await this.eCertFullTimeService.getResponseFilesFullPath(
+      this.esdcConfig.ftpResponseFolder,
+      new RegExp(
+        `^${this.esdcConfig.environmentCode}EDU.PBC.CERTSFB.*\\.DAT`,
+        "i",
+      ),
+    );
+    const processFiles: ProcessSftpResponseResult[] = [];
+    for (const filePath of filePaths) {
+      processFiles.push(await this.processFile(filePath));
+    }
+    return processFiles;
+  }
+
+  /**
+   * Process each individual E-Cert response file from the SFTP.
+   * @param filePath E-Cert response file to be processed.
+   * @returns Process summary and errors summary.
+   */
+  private async processFile(
+    filePath: string,
+  ): Promise<ProcessSftpResponseResult> {
+    const result = new ProcessSftpResponseResult();
+    result.processSummary.push(`Processing file ${filePath}.`);
+
+    let responseFile: ECertResponseRecord[];
```

Comment:
Just for clarification, what happens if the e-cert response does not have any errors?

Reply:
this file will only have records that have error codes for a document number, if a document number does not have an error code, then the record won't be there in the file, and also the trailer line has the total error record count. before processing the error code records, we are checking 
![image](https://user-images.githubusercontent.com/77353155/144463102-4ee52fcc-11c0-4557-b7ca-c6e0b3a2b60c.png)
, if there is a mismatch, then we won't process the file. hope this answers your question, else we can sync up.